### ISSUE TITLE
Editor 2 crashes when importing (some) projects. [DEFEDIT-4164]

### ISSUE THREAD

**Issue by AxelHammarback** [defold/editor2-issues#2443]
### Expected behaviour
I expect project to be imported successfully.

### Actual behaviour
I get the following error:

"org.eclipse.jgit.api.errors.TransportException: http://cr.defold.se:9998/prjs/1828: git-upload-pack not permitted
org.eclipse.jgit.errors.TransportException: http://cr.defold.se:9998/prjs/1828: git-upload-pack not permitted"

Followed (sometimes) by a hard crash. 

The error doesn't happen to **all** projects, but for ~50% of the ones I've tried. 

Also, it's a bit annoying that the folder gets created even if the import fails. It's cumbersome to have to manually delete the folder which was created, if I want to try to import again.

### Steps to reproduce
See attached screen capture. Is this the dreaded Mojave bug people are talking about?

[defold-crash.mov.zip](https://github.com/defold/editor2-issues/files/2828677/defold-crash.mov.zip)

<hr/>

<table>
<tr><td>Build time</td><td>2019-01-28T15:36:49.949805</td></tr>
<tr><td>Defold channel</td><td>editor-alpha</td></tr>
<tr><td>Defold editor sha</td><td>c043948fbf2c57e88d4ca921ea9ab135081bb7e7</td></tr>
<tr><td>Defold engine sha</td><td>9f376d88408bad9088a4187d4f874ae4767a0185</td></tr>
<tr><td>Defold version</td><td>1.2.145</td></tr>
<tr><td>Error</td><td><a href='https://sentry.io/defold/editor2/?query=id%3A"36ccd22a124540b7a8e466032671a2c8"'>36ccd22a124540b7a8e466032671a2c8</a></td></tr>
<tr><td>GPU</td><td>AMD Radeon Pro 560X OpenGL Engine</td></tr>
<tr><td>GPU Driver</td><td>2.1 ATI-2.4.9</td></tr>
<tr><td>Java version</td><td>1.8.0_102-b14</td></tr>
<tr><td>OS arch</td><td>x86_64</td></tr>
<tr><td>OS name</td><td>Mac OS X</td></tr>
<tr><td>OS version</td><td>10.14.2</td></tr>
<table>

**Comment by K-Klear** [comment]
I just ran into this when importing a project: 

> org.eclipse.jgit.api.errors.TransportException: http://cr.defold.se:9998/prjs/97204: cannot open git-upload-pack
> org.eclipse.jgit.errors.TransportException: http://cr.defold.se:9998/prjs/97204: cannot open git-upload-pack
> java.net.UnknownHostException: cr.defold.se

I am in a place with a very spotty connection though. It usually takes a few tries to synchronize successfully ("Error getting changes" is the usual message, I think). This is probably the same issue but when importing.

One more thing - the error message had "dismiss" and "report" buttons, neither of which seemed to do anything. Closing it with the X in the top right corner produced a new error:

> java.lang.AssertionError: Assert failed: (c/cache? cache)

Actually, this one popped up about six times - I think there was one for every press of one or both of the buttons in the previous error message, but I can't confirm that, as the import was successful on my next try.

And yeah, the failed import created a folder with .git in it, which has to be deleted to try again, which is annoying.
